src/common/components/dateTimeTextField/DateTimeTextField.tsx

```diff
@@ -12,52 +12,69 @@ const useStyles = makeStyles({
   textFieldInput: { width: '100%' },
 });
 
+/**
+ * Removes the special prefix `@@react-admin@@` and the needless quotation marks
+ * that might be added by the useInput-hook.
+ * @param message the translation key (with ra-core's special prefix to avoid double translation)
+ * @returns the fixed translation key
+ * @example
+ *    // a code related to the validation in the useInput.ts...
+ *    validate: async (value, values) => {
+ *        if (!sanitizedValidate) return true;
+ *        const error = await sanitizedValidate(value, values, props);
+ *        if (!error) return true;
+ *        // react-hook-form expects errors to be plain strings but our validators can return objects
+ *        // that have message and args.
+ *        // To avoid double translation for users that validate with a schema instead of our validators
+ *        // we use a special format for our validators errors.
+ *        // The ValidationError component will check for this format and extract the message and args
+ *        // to translate.
+ *        return `@@react-admin@@${JSON.stringify(error)}`;
+ */
+const handleUseInputDoubleValidationMessage = (message: string) =>
+  message.replace('@@react-admin@@', '').replaceAll('"', '');
+
 export type BoundedTextFieldProps = InputProps & TextFieldProps;
 
 const BoundedTextField = (props: BoundedTextFieldProps) => {
   const translate = useTranslate();
   const classes = useStyles();
-
-  const { label, variant, defaultValue, disabled } = props;
+  const { validate, ...rest } = props;
   const {
-    field: { name, onChange },
+    field,
     fieldState: { isTouched, error },
     formState: { isSubmitted },
     isRequired,
-  } = useInput(props);
+  } = useInput({ validate, ...rest });
 
   return (
     <TextField
-      variant={variant}
+      {...field}
       className={classes.textFieldInput}
       size="small"
-      name={name}
-      label={label}
-      onChange={onChange}
       error={!!(isTouched && error)}
       helperText={
         (isTouched || isSubmitted) && !!error
-          ? translate(error?.message ?? '')
+          ? translate(
+              handleUseInputDoubleValidationMessage(error?.message ?? '')
+            )
           : undefined
       }
       required={isRequired}
-      defaultValue={defaultValue}
-      disabled={disabled}
+      {...rest}
```

Comment:
As `rest` comes from `Omit<BoundedTextFieldProps, 'validate'>` it already has `required` in it through `BaseTextFieldProps` which it extends, so `{...rest}` can override `required={isRequired}` here. Is this intentional?

Reply:
`{...rest}` also overrides these from `field`:
 - `onBlur`
 - `disabled`
 - `value`
 - `name`
because all these are properties of `BaseTextFieldProps` which `BoundedTextFieldProps` extends.

`onChange` and `ref` in `field` are probably overridden also by `{...rest}` as these showed some types:
```
type TestType1= Omit<BoundedTextFieldProps, 'validate'>['onChange'];
type TestType2 = Omit<BoundedTextFieldProps, 'validate'>['ref'];
```